### DISCUSSION TITLE
How to upload eye-tracking experiment on cognition platform?

### DISCUSSION THREAD

**Discussion by zaizibai** [jspsych/jsPsych#1936] [Q&A] [answered]
Dear expert in jspsych,
Recently webgazer extension was introduced into jspsych, and I plan to use it to perform a online eye-tracking experiment.However, afterI upload my js file into cognition platform, there is some error feedback and experiment cannot be properly carry out.

**Comment by GEJ1** [comment, marked as answer]
Hi @zaizibai , 

It would be great if you share the error, but based in my experience using cognition and the webgazer extension you can check this points:

1. Make sure you are uploading the [jspsych fork of webgazer.js](https://github.com/jspsych/WebGazer) as you can see in the jspsych [overview for eye tracking](https://www.jspsych.org/overview/eye-tracking/).
2. Check jspsych version is 6.3.1
3. Normally it's necesary to use at least some of the jspsych plugins to interact with webgazer: jspsych‑webgazer‑calibrate, jspsych‑webgazer‑init‑camera, jspsych‑webgazer‑validate .

I hope this helps you,

Best,

Gustavo.

**Reply by zaizibai** [reply]
Thank you for responding, I have uploaded the webgazer library, but still, the experiment can not be carried out.  The error that cognition returns is 'The task is not ready yet..'

**Reply by GEJ1** [reply]
Can you paste your code here? at least the part related with eye tracking.

Also it would be useful if you open the link of the experiment and then press F12 (in chrome) and copy here the error that appears in the console.

edit: I think the way that works for me was to use the webgazer.js file that @jodeleeuw use in his eyetrack example: https://github.com/jodeleeuw/219-eyetrack-example-full/blob/main/experiment/webgazer/webgazer.js
I was thinking that this same file was in the webgazer jspsych fork but isn't. That is correct @jodeleeuw ?

**Reply by zaizibai** [reply]
Thank you for providing help. It turns out it is my stupid mistake that I forgot to upload my css file.....

**Reply by zaizibai** [reply]
Apart from that, I have a question about the webgazer data that the extension returns . Is the base point of data located at the left-down corner? What is the unit of the coordinate system, I let one of my friends do the task, and CSS return his monitor resolution  1280*960, however, some data's X axis exceeds 1280...Thank you very much.
